src/plugins/kibana_react/public/theme/kibana_theme_provider.tsx

```diff
@@ -48,7 +50,7 @@ export const KibanaThemeProvider: FC<KibanaThemeProviderProps> = ({ theme$, modi
       utilityClasses={false}
       modify={modify}
     >
-      {children}
+      <EmotionThemeProvider theme={euiTheme}>{children}</EmotionThemeProvider>
```

Comment:
FYI, this theme provider will only work at the top level and will need to be repeated for any team/plugin that reuses a nested `<EuiThemeProvider>` to set granular theming or color modes. For example:

```tsx
import { EuiThemeProvider } from '@elastic/eui';

// Top level of app
<KibanaThemeProvider>
  // the `theme` prop vars will be correct here

  <EuiThemeProvider colorMode="inverse">
    // the `theme` prop vars will NOT be correct here
    // and consumers will need to re-nest an <EmotionThemeProvider>
  </EuiThemeProvider>
</KibanaThemeProvider>
```

For this reason, I'm tempted to bake Emotion's `ThemeProvider` into EUI by default (https://github.com/elastic/eui/pull/6913) because consumers will generally always need to use EUI's `<EuiThemeProvider>` to customize component-level theming or color modes.

The main downside to doing so would be a loss of flexibility, e.g. if a dev wants to pass in their own custom Emotion theme 🤔 

Would really appreciate your thoughts on this and what level you think makes the most sense to tackle this at (EUI-level or consumer/Kibana level)!

Reply:
Thanks for the feedback, really appreciate it :) 

I usually am undecided when it comes to providing the ''working'' solution vs ''giving flexibility'' :PIn this case I would aim at unification of styles instead of providing custom styles per plugin. But that's just my opinion. 
 Do we know if any users (are using / want to use) custom themes?  Because if they don't - we have a problem solved ;)

Reply:
Custom theming is a bit of an edge case, per-component color mode is the more common usage of `EuiThemeProvider`. I just did a quick grep and here's the ones that I found that aren't top-level / app-level usages:

- [serverless_search - code block](https://github.com/elastic/kibana/blob/67fc8333e75b3035bd5ed984c04730aabef36526/x-pack/plugins/serverless_search/public/application/components/code_box.tsx#L75)
- [serverless_search - api key](https://github.com/elastic/kibana/blob/67fc8333e75b3035bd5ed984c04730aabef36526/x-pack/plugins/serverless_search/public/application/components/api_key/api_key.tsx#L188)
- [core chrome - collapsible nav](https://github.com/elastic/kibana/blob/67fc8333e75b3035bd5ed984c04730aabef36526/packages/core/chrome/core-chrome-browser-internal/src/ui/header/collapsible_nav.tsx#L174)
- [enterprise_search - code block](https://github.com/elastic/kibana/blob/67fc8333e75b3035bd5ed984c04730aabef36526/x-pack/plugins/enterprise_search/public/applications/vector_search/components/dev_tools_console_code_block/dev_tools_console_code_block.tsx#L51)
- [serverless_search - integration panel](https://github.com/elastic/kibana/blob/67fc8333e75b3035bd5ed984c04730aabef36526/x-pack/plugins/serverless_search/public/application/components/overview_panels/integrations_panel.tsx#L27)

Most appear to be within Enterprise Search, and most also appear to not use any custom CSS nested, so... right now it's fairly safe, but it's always a question of what consumers choose to do later and whether it causes us support time down the road 😅

It's useful to hear you don't have a strong opinion either way - let me see what the rest of the EUI team thinks and get back to you!

Reply:
Oh wait sorry, I totally misread, I thought you were asking about custom component theme styles rather than custom Emotion themes. There's been sorta one request from a non-Elastic consumer: https://github.com/elastic/eui/discussions/5351

But in general since Kibana is our primary customer so to speak, we do try to prioritize the developer experience here generally 🤔